python/dgl/mock_sparse/reduce.py

```diff
@@ -0,0 +1,286 @@
+import torch
+
+from .sp_matrix import SparseMatrix
+
+
+def reduce(A: SparseMatrix, dim=None, rtype: str = 'sum'):
+    """Reduce sparse matrix A on the dimension `dim`.
+
+    If `dim` is None, it reduces all the elements in the sparse
+    matrix. Otherwise, it reduces on the row (dim=0) or column (dim=1) dimension,
+    producing a tensor of shape `(A.shape[1], ) + A.val.shape[:1]` or
+    `(A.shape[0],) + A.val.shape[:1]`.
+
+    The reduction does not count zero values. If the row or column to be reduced
+    does not have any non-zero value, the result will be 0.
+
+    Parameters
+    ----------
+    A : SparseMatrix
+        Sparse matrix
+    dim : int, optional
+        The dimension to reduce.
+    rtype: str
+        Reduction type, one of ['sum', 'smin', 'smax', 'mean']
```

Comment:
Why we have `smin` and `smax`, not `ssum` and `smean`?

Reply:
Right. I added `smean` but not `ssum` because `0` does not affect the result.